### PR TITLE
Adding local_port to port forward, fix missing image_name for docker_build

### PR DESCRIPTION
**Pull request by cesarfm**
- adding optional `local_port` option for `gke_port_forward`, this is needed because some flows port-forward to more than one service
- fixing bug in `docker_build_push`: it had image name hardcoded to `search-solr`

Example of usage:
- https://github.com/ricardo-ch/search-solr-admin/pull/48

### CODE REVIEW

File: src/commands/docker_build_push.yml [hunk 1]
```diff
@@ -2,14 +2,16 @@ description: >
...
     default: .
 
 steps:
   - run:
       name: 'docker build'
-      command: docker build -t ric-docker.jfrog.io/ricardo-ch/search-solr:${CIRCLE_BRANCH}-${CIRCLE_SHA1} .
+      command: docker build -t ric-docker.jfrog.io/ricardo-ch/<< parameters.image_name >>:${CIRCLE_BRANCH}-${CIRCLE_SHA1} .
   - run:
       name: 'docker push'
-      command: docker push ric-docker.jfrog.io/ricardo-ch/search-solr:${CIRCLE_BRANCH}-${CIRCLE_SHA1}
```
Comment (michaelkuechler) [inline src/commands/docker_build_push.yml:15]: 🤦‍♂️  great job MiK 🤦‍♂️

File: src/commands/gke_port_forward.yml [hunk 2]
```diff
@@ -7,15 +7,35 @@ parameters:
...
     type: string
 steps:
-  - run:
-      background: true
-      name: Setup port forwarding to target
-      command: kubectl port-forward --context <<parameters.context>> -n <<parameters.namespace>> <<parameters.target>> <<parameters.port>>
-  - run:
-      name: Wait for port forwarding setup completion
-      command: dockerize -wait tcp://localhost:<<parameters.port>> -timeout 60s
+  # if local_port should be the same than port
```
Comment (michaelkuechler) [inline src/commands/gke_port_forward.yml:18]:
```suggestion
  # if local_port should be the same as port
```

File: src/commands/README.md [hunk 3]
```diff
@@ -101,6 +103,7 @@ steps:
 **Parameters**:
 - **target**: service/pod to which traffic is forwarded
 - **port**: service/pod port to portforward
+- **local_port**: (optional) service/pod port to portforward on the LOCAL (CircleCI) side. If missing, it is the same than port
```
Comment (michaelkuechler) [inline src/commands/README.md:106]:
```suggestion
- **local_port**: (optional) service/pod port to portforward on the LOCAL (CircleCI) side. If missing or 0, it is the same than port
```

File: src/commands/gke_port_forward.yml [hunk 4]
```diff
@@ -7,15 +7,35 @@ parameters:
...
+        equal: [ 0, <<parameters.local_port>> ]
+      steps:
+        - run:
+            background: true
+            name: Setup port forwarding to target
+            command: kubectl port-forward --context <<parameters.context>> -n <<parameters.namespace>> <<parameters.target>> <<parameters.port>>
+        - run:
+            name: Wait for port forwarding setup completion
+            command: dockerize -wait tcp://localhost:<<parameters.port>> -timeout 60s
+  # if local_port has a value
```
Comment (michaelkuechler) [inline src/commands/gke_port_forward.yml:30]:
```suggestion
  # if local_port has a value, use it
```

File: src/commands/gke_port_forward.yml [hunk 5]
```diff
@@ -7,15 +7,35 @@ parameters:
...
-      background: true
-      name: Setup port forwarding to target
-      command: kubectl port-forward --context <<parameters.context>> -n <<parameters.namespace>> <<parameters.target>> <<parameters.port>>
-  - run:
-      name: Wait for port forwarding setup completion
-      command: dockerize -wait tcp://localhost:<<parameters.port>> -timeout 60s
+  # if local_port should be the same than port
+  - when:
+      condition:
+        equal: [ 0, <<parameters.local_port>> ]
```
Comment (michaelkuechler) [inline src/commands/gke_port_forward.yml:21]: maybe we should check for p<=0  🤔

**Review by michaelkuechler — APPROVED** [review]
some small comments

id personally change the condition to prevent unexpected behaviour for port<0

File: src/commands/gke_port_forward.yml (hunk 5 above)
[the remark this answers is not in the document]
Comment (cesarfm) [reply src/commands/gke_port_forward.yml:21]: Sadly CircleCI does not support numerical comparison operators 😞   so this is "good enough".

**Pull request merged**